Fireworks/database.py

```diff
@@ -3,19 +3,19 @@
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import sessionmaker, Query
 from Fireworks import Message, cat
-from Fireworks.source import Source, PassThroughSource
+from Fireworks.pipeline import Pipe, PassThroughPipe
 import numpy as np
 import pandas as pd
 
 Base = declarative_base()
 
-class TableSource(PassThroughSource):
+class TablePipe(PassThroughPipe):
     """
-    Represents an SQLalchemy Table while having the functionality of a Source.
+    Represents an SQLalchemy Table while having the functionality of a Pipe.
     """
-    def __init__(self, table, engine, columns = None, inputs = None, **kwargs):
+    def __init__(self, table, engine, columns = None, input = None, **kwargs):
 
-        super().__init__(inputs=inputs, **kwargs)
+        super().__init__(input=input, **kwargs)
         Session = sessionmaker(bind=engine)
```

Comment:
I don't think it's a good idea to open a sqlalchemy session inside the class instance. Typically you want session management outside and independent of data operations. Also I do don't see a method to close a session, which leaves the risk of having open threads. Consider making session an input to TableSource because the TableSource shouldn't be managing the session or if you want to keep engine an input have the engine execute the query directly like

```
connection = engine.connect()
result = connection.execute("select username from users")
for row in result:
    print("username:", row['username'])
connection.close()
```

Reply:
from the sqlalchemy docs 
```
tl;dr;

As a general rule, keep the lifecycle of the session separate and external from functions and objects that access and/or manipulate database data. This will greatly help with achieving a predictable and consistent transactional scope.

Make sure you have a clear notion of where transactions begin and end, and keep transactions short, meaning, they end at the series of a sequence of operations, instead of being held open indefinitely.
```